docs/notebook-demos/layers.ipynb

```diff
@@ -0,0 +1,1310 @@
+    "\n",
+    "`collect_metadata` will return the `Metadata` of the layer that fits the `layout` given."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Collecting Metadata with the default LocalLayout()\n",
+    "metadata = raster_layer.collect_metadata()\n",
+    "metadata"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Collecting Metadata with the default GlobalLayout()\n",
+    "raster_layer.collect_metadata(layout=gps.GlobalLayout())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Collecting Metadata with a LayoutDefinition\n",
+    "extent = gps.Extent(0.0, 0.0, 33.0, 33.0)\n",
+    "tile_layout = gps.TileLayout(2, 2, 256, 256)\n",
+    "layout_definition = gps.LayoutDefinition(extent, tile_layout)\n",
+    "\n",
+    "raster_layer.collect_metadata(layout=layout_definition)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Reproject\n",
+    "\n",
+    "`reproject` will change the projection the rasters within the layer to the given `target_crs`. This method does not sample past the tiles' boundaries."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# The CRS of the layer before reprojecting\n",
+    "metadata.crs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# The CRS of the layer after reprojecting\n",
+    "raster_layer.reproject(target_crs=3857).collect_metadata().crs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Tiling Data to a Layout\n",
+    "\n",
+    "`tile_to_layout` will tile and format the rasters within a ``RasterLayer`` to a given layout. The result of this tiling is a new instance of `TiledRasterLayer`. This output contains the same data as its source `RasterLayer`, however, the information contained within it will now be orginized and assorted to the given layout.\n",
+    "\n",
+    "During this step it is also possible to reproject the `RasterLayer`. This can be done by specifying the `target_crs` to reproject to. Reprojecting using this method produces a different result than what is returned by the `reproject` method. Whereas the latter does not sample past the boundaries of rasters within the layer, the former does. This is important as anything with a `GlobalLayout` needs to sample past the boundaries of the rasters."
```

Comment:
This is a bit confusing.  If it hasn't been explained elsewhere, may want to put in a bit more detail, if it has, add a link.